agentstack/generation/tool_generation.py

```diff
@@ -1,411 +1,172 @@
+            tools_init.add_import_for_tool(framework, tool)
+    except ValidationError as e:
+        print(term_color(f"Error adding tool:\n{e}", 'red'))
 
-    if tool_data.packages:
-        packaging.install(' '.join(tool_data.packages))
-    shutil.copy(tool_file_path, f'{path}src/tools/{tool_name}_tool.py')  # Move tool from package to project
-    add_tool_to_tools_init(tool_data, path)  # Export tool from tools dir
-    add_tool_to_agent_definition(
-        framework=framework, tool_data=tool_data, path=path, agents=agents
-    )  # Add tool to agent definition
+    # Edit the framework entrypoint file to include the tool in the agent definition
+    if not agents:  # If no agents are specified, add the tool to all agents
+        agents = frameworks.get_agent_names(framework, path)
+    for agent_name in agents:
+        frameworks.add_tool(framework, tool, agent_name, path)
 
-    if tool_data.env:  # add environment variables which don't exist
+    if tool.env:  # add environment variables which don't exist
         with EnvFile(path) as env:
-            for var, value in tool_data.env.items():
+            for var, value in tool.env.items():
                 env.append_if_new(var, value)
         with EnvFile(path, filename=".env.example") as env:
-            for var, value in tool_data.env.items():
+            for var, value in tool.env.items():
                 env.append_if_new(var, value)
 
-    if tool_data.post_install:
-        os.system(tool_data.post_install)
+    if tool.post_install:
+        os.system(tool.post_install)
 
     with agentstack_config as config:
-        config.tools.append(tool_name)
+        config.tools.append(tool.name)
 
-    print(term_color(f'🔨 Tool {tool_name} added to agentstack project successfully', 'green'))
-    if tool_data.cta:
-        print(term_color(f'🪩 {tool_data.cta}', 'blue'))
+    print(term_color(f'🔨 Tool {tool.name} added to agentstack project successfully', 'green'))
+    if tool.cta:
+        print(term_color(f'🪩  {tool.cta}', 'blue'))
 
 
-def remove_tool(tool_name: str, path: Optional[str] = None):
-    if path:
-        path = path.endswith('/') and path or path + '/'
-    else:
-        path = './'
-
-    framework = get_framework()
+def remove_tool(tool_name: str, agents: Optional[list[str]] = [], path: Optional[Path] = None):
+    if path is None:
+        path = Path()
     agentstack_config = ConfigFile(path)
+    framework = agentstack_config.framework
 
     if tool_name not in agentstack_config.tools:
         print(term_color(f'Tool {tool_name} is not installed', 'red'))
         sys.exit(1)
 
-    tool_data = ToolConfig.from_tool_name(tool_name)
-    if tool_data.packages:
-        packaging.remove(' '.join(tool_data.packages))
+    tool = ToolConfig.from_tool_name(tool_name)
+    if tool.packages:
+        packaging.remove(' '.join(tool.packages))
+
+    # TODO ensure that other agents in the project are not using the tool.
```

Comment:
this should already be handled with this code, right?

Reply:
That just uninstalls the dependencies. 

#76 expands that to handle copying/deleting the tool source files and installing/removing deps.

Reply:
Actually in #76 I think we should leave deps installed and just delete the tool src files.